### PR TITLE
Lecture2 c

### PR DESCRIPTION
**Pull request by SehedaOleh**
All home tasks are done. couple probably can be optimized but I didn't have time to do it now. I will be appreciate you for some feedback.

### CODE REVIEW

File: Lecture_2/Ex1.c [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    printf ("Number in uint16 befor swap%X\n", init_uint16_number);
+    printf ("Number in uint16 after swap%X\n", swap_result_uint16);
+    
+    //-------------------------------------------------------------------
+    uint32_t init_uint32_number = 0xC2F493A1;
+    uint32_t swap_result_uint32 = 0x00000000;
+    
+    swap_result_uint32 = swap32(init_uint32_number);
+    
+    printf ("Number in uint32 befor swap%X\n", init_uint32_number);
```
Comment (rikorsev) [inline Lecture_2/Ex1.c]: Output would be more readable if you specify that output number shows in HEX format. So, I would change **all printf** string to the following:
printf("Number in uint32 befor swap 0x%X\n", init_uint32_number);

File: Lecture_2/Ex1.c [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    
+    printf ("Number in uint64 befor swap%lX\n", init_uint64_number);
+    printf ("Number in uint64 after swap%lX\n", swap_result_uint64);
+    return 0;
+}
+uint16_t swap16 (uint16_t input16_number)
+{
+    uint16_t first_byte = 0x0, secound_byte = 0x0, result16 = 0x0;
+    
+    first_byte = input16_number<<8&0xFF00;
```
Comment (rikorsev) [inline Lecture_2/Ex1.c]: Code became more readable if you put whitespaces around operators, like this:
first_byte = input16_number << 8 & 0xFF00;

File: Lecture_2/Ex4.c [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+ Write a C program to compute the perimeter and 
+ area of a circle with a radius of 6 meters;
+
+*******************************************************************************/
+
+#include <stdio.h>
+#include <stdint.h>
+
+#define M_PI 3.14159265358979323846
```
Comment (rikorsev) [inline Lecture_2/Ex4.c]: **M_PI** is defined in <math.h> include file. It is unnecessary to define it here

File: Lecture_2/Ex5.c [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    uint32_t result = 1;
+   
+    for (int i = 1; i <= fact_base_number; i++)
+    {
+       result*=i;
+    }
+    
+    printf("Factorial of %d is %d", fact_base_number, result);
+    return 0;
+}
```
Comment (rikorsev) [inline Lecture_2/Ex5.c:20]: In general program looks fine. But I would prefer that you write common function to calculate factorials to use it in the future.

File: Lecture_2/Ex6.c [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+int main()
+{
+    int user_number = 1;
+    int start = 1, stop = 500;
+    do {
+        printf("Please insert number betwen %d and %d\n",start,stop);
+        scanf("%d", &user_number);
+    }
+    while (user_number > 500 && user_number < 1);
```
Comment (rikorsev) [inline Lecture_2/Ex6.c]: Good! You are checking a range!

File: Lecture_2/Ex6.c [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    
+    for(int i = start - 1; i < stop; i++)
```
Comment (rikorsev) [inline Lecture_2/Ex6.c]: Why do you specify here i = start - 1?
I thing here is just enough to write i = start;

**Comment by rikorsev** [comment]
In general looks very good! Code style looks accurate and smooth. 
I'm very like that you divide your code by commits. It's really making code review simpler.

Just one think that I would like to mention:
Usually, when you start to work on new feature set, you should make a branch **from master** but **not from lecture_1**

**Comment by SehedaOleh** [comment]
Added some changes to this branch according to your comments.

**Comment by rikorsev** [comment]
Every thing looks fine. Nice job! Now you could merge it to master branch.

**Pull request merged**